File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FLinearLayout.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.Color;
import android.util.Log;
import android.view.View;
import android.widget.LinearLayout;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FLinearLayout extends FView implements AttrSettable,AttrGettable {
    private LinearLayout v;
    public FLinearLayout(UIController c){
        parentController =c;
        v = new LinearLayout(parentController.activity);
        view=v;
        v.setOrientation(LinearLayout.VERTICAL);
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
                // --------------------------------------------------
            case "Orientation":
                if (v.getOrientation()==LinearLayout.HORIZONTAL)
                    return "HORIZONTAL";
                else
                    return "VERTICAL";
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
                // ----------------------------------------------------------------------------
            case "Orientation":
                if (value.equals("HORIZONTAL")) {
                    v.setOrientation(LinearLayout.HORIZONTAL);
                }else {
                    v.setOrientation(LinearLayout.VERTICAL);
                }
                break;
            case "AddView":
                final String childVid=value;
                FView f=parentController.viewmap.get(childVid);
                v.addView(f.view,parseLP(f));
                break;
            case "AddViewAt":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    int pos = Integer.parseInt(array.getString(0));
                    String vid = array.getString(1);
                    FView f1 = parentController.viewmap.get(vid);
                    v.addView(f1.view,pos,parseLP(f1));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "OnClick":
                v.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Faithdroid.triggerEventHandler(value,"");
                    }
                });
                break;
        }
    }

    public static LinearLayout.LayoutParams parseLP(FView f) {
        LinearLayout.LayoutParams lp = new LinearLayout.LayoutParams(f.size[0], f.size[1]);
        lp.gravity=f.layoutGravity;
        lp.weight=f.layoutWeight;
        lp.leftMargin = f.margin[0];
        lp.topMargin = f.margin[1];
        lp.rightMargin = f.margin[2];
        lp.bottomMargin = f.margin[3];
        return lp;
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FLinearLayout.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.widget.LinearLayout
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.UIController
import org.json.JSONArray
import org.json.JSONTokener


class FLinearLayout(c: UIController) : FView(), AttrSettable, AttrGettable {
    private val v: LinearLayout

    init {
        parentController = c
        v = LinearLayout(parentController!!.activity)
        view = v
        v.orientation = LinearLayout.VERTICAL
    }

   override fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            // --------------------------------------------------
            "Orientation" -> return if (v.orientation == LinearLayout.HORIZONTAL)
                "HORIZONTAL"
            else
                "VERTICAL"
        }
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // ----------------------------------------------------------------------------
            "Orientation" -> if (value == "HORIZONTAL") {
                v.orientation = LinearLayout.HORIZONTAL
            } else {
                v.orientation = LinearLayout.VERTICAL
            }
            "AddView" -> {
                val f = parentController!!.viewmap.get(value)
                v.addView(f!!.view, parseLP(f))
            }
            "AddViewAt" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                val pos = Integer.parseInt(array.getString(0))
                val vid = array.getString(1)
                val f1 = parentController!!.viewmap.get(vid)
                v.addView(f1!!.view, pos, parseLP(f1!!))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "OnClick" -> v.setOnClickListener { Faithdroid.triggerEventHandler(value, "") }
        }
    }

    companion object {

        fun parseLP(f: FView): LinearLayout.LayoutParams {
            val lp = LinearLayout.LayoutParams(f.size[0], f.size[1])
            lp.gravity = f.layoutGravity
            lp.weight = f.layoutWeight
            lp.leftMargin = f.margin[0]
            lp.topMargin = f.margin[1]
            lp.rightMargin = f.margin[2]
            lp.bottomMargin = f.margin[3]
            return lp
        }
    }
}
